src/sec.c

```diff
@@ -23,11 +23,10 @@ sec_bin_init(sec_bin_t *bin) {
 bool
 sec_init(tsdn_t *tsdn, sec_t *sec, base_t *base, pai_t *fallback,
     const sec_opts_t *opts) {
-	size_t max_alloc = opts->max_alloc & PAGE_MASK;
-	pszind_t npsizes = sz_psz2ind(max_alloc);
-	if (sz_pind2sz(npsizes) > opts->max_alloc) {
-		npsizes--;
-	}
+	assert((opts->max_alloc & PAGE_MASK) == 0);
```

Comment:
I was thinking that if we deal with extents here, it would be easier to force `max_alloc` to be a PAGE multiple, but thinking more about it, you're right. `max_alloc` could still be a PAGE multiple, but fall between two size classes, so this restriction doesn't help.